CIP-0119/README.md

```diff
@@ -0,0 +1,276 @@
+
+### Extended Vocabulary
+Like CIP-108, this CIP also extends the potential vocabulary of CIP-100's `body` property. 
+
+Furthermore we extend the Schema.org definition of a [Person](https://schema.org/Person). Any property of Person maybe included within the `body`.
+
+>**Reminder for tooling providers/builders** DRep metadata is user generated content.
+
+The following are a list of properties tooling should expect to encounter:
+
+#### `paymentAddress`
+- Optional
+- Bech32 encoded address, for the same network as the DRep registration is to be submitted to.
+
+Dreps may want to recieve tokens for a variety of reasons such as:
+- donations
+- expenses
+- any incentive program
+
+Therefore there MAY be a `paymentAddress` field in the metadata where such payments could be sent. This makes such an address public and payments to DReps transparent. 
+
+This SHOULD NOT be confused with the `address` property of a [Person](https://schema.org/Person), `address` in the context of a DRep refers to their location and NOT their payment address.
+
+#### `givenName`
+- Compulsory
+- This is a property inherited from [Person](https://schema.org/Person)
+- It is the only compulsory property
+- A very short freeform text field. Limited to 80 characters.
+- This MUST NOT support markdown text styling.
+- It is intended that authors will use this field for their profile name/ username.
+
+#### `image`
+- Optional
+- This is a property inherited from [Person](https://schema.org/Person)
+- This SHOULD be treated as the profile picture of the individual
+- This MUST contain a fully described [`imageObject`](https://schema.org/ImageObject) property as opposed to a URL. 
+ 
+##### `imageObject`
+- This is to be included in a metadata file as a property of the `image` property, only if the `image` property is included.
+- It explains the image to those (inc. tools) who are viewing it.
+- `imageObject` MUST contain a base64 encoded image in its [`contentURL`](https://github.com/schemaorg/schemaorg/issues/2696) property in a [dataURI](https://en.wikipedia.org/wiki/Data_URI_scheme) format:
+  - i.e. _data:content/type;base64,_ (AND NOT _data:domain.tld_)
+  - e.g. _contentURL:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAUAAAAFCAYAAACNbyblAAAAHElEQVQI12P4//8/w38GIAXDIBKE0DHxgljNBAAO9TXL0Y4OHwAAAABJRU5ErkJggg==_ (AND NOT _contentURL:https://avatars.githubusercontent.com/u/113025685?v=4_)
```

Comment:
I dislike this requirement that the image must be embedded within the metadata file itself as this will quickly lead to relatively massive (multi-megabyte) metadata files that are more difficult to fetch and store without gaining substantial value.

The schema.org `imageObject` already provides for a `sha256` field to provide a confirmation and validation hash that the asset in question is unmodified from the time of submission and allows for offloading. This will become particularly important when (most likely in the future) we have groups of dReps who all have a profile photo or we have proposals that need quite a few different images as citation and reference. 

It is more logical to develop a strategy to work with "remote" files rather than require that they all be b64 encoded and included directly in the metadata file itself.

Reply:
I would modify the language such:

**If the `imageObject` DOES NOT contain a base64 encoded image, the `sha256` property MUST be populated and match the contents of the image file found at the `contentUrl`**

Reply:
I get where you are coming from, have you considered:

- The point of having external metadata files (e.g. on IPFS) was that people could store more information without congesting Cardano
- Introduction of another hashing algorithm other than blake 2b could be confusing
- Another method of saving images would add complexity for tooling providers who are building tools that want to show these images
  - If tools reading the metadata need to visit the url link every time they show an image this would take ages for the image to load. 
  - Alternatively they could save images and only show them if the SHA256 hash matches 
- If tools want to implement a porn filter b64 images only need to be checked once whereas an image at a url might change and so would need to be checked every time they are opened. This is somewhat ameliorated by the hash check, but again this is more overhead for tooling.

?

Reply:
On the other hand CIP-100 does allow for metadata to be saved as part of the governance transaction so maybe your suggestion is necessary for such instances. I'm keen for a discussion.

Reply:
@Crypto2099 , @rphair , @Ryun1  I have made changes in line with the above comments.